openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
+            if fission.shape == shape:
+                self._fission = np.copy(fission)
+                if np.sum(self._fission) > 0.0:
+                    self._fissionable = True
+            else:
+                msg = 'Shape of provided fission "{0}" does not match shape ' \
+                      'required, "{1}"'.format(fission.shape, shape)
+                raise ValueError(msg)
+
+    def set_nu_fission(self, nu_fission, **kwargs):
+        # The NuFissionXS class does not have the capability to produce
+        # a fission matrix and therefore if this path is pursued, we know
+        # chi must be used.
+        if isinstance(nu_fission, openmc.mgxs.NuFissionXS):
+            # Make sure passed MGXS object contains correct group structure
+            if self.energy_groups != nu_fission.energy_groups:
+                msg = 'Group structure of provided NuFissionXS does not match'\
+                      ' group structure of XSdata object'
+                raise ValueError(msg)
+            # Get openmc.mgxs.get_xs() arguments from kwargs
+            # nuclides, xs_type, and value will have sane defaults but can be
+            # overridden by kwards
+            if 'nuclides' in kwargs:
+                nuclides = kwargs['nuclides']
+            else:
+                nuclides = 'sum'
+            if 'xs_type' in kwargs:
+                xs_type = kwargs['xs_type']
+            else:
+                xs_type = 'macro'
+            if 'value' in kwargs:
+                value = kwargs['value']
+            else:
+                value = 'mean'
+            # subdomains is required from the kwargs as this is specific to
+            # this XSdata object.
+            if 'subdomains' in kwargs:
+                subdomains = kwargs['subdomains']
+            else:
+                msg = "Argument 'subdomains' is required"
+                raise ValueError(msg)
+
+            if self._representation is 'isotropic':
+                self._nu_fission = nu_fission.get_xs(subdomains=subdomains,
+                                                     nuclides=nuclides,
+                                                     xs_type=xs_type,
+                                                     value=value)
+            elif self._representation is 'angle':
+                # Not yet implemented as MGXS do not yet support this
+                pass
+
+            self._use_chi = True
+
+        else:
+            # nu_fission can be given as a vector or a matrix
+            # Vector is used when chi also exists.
+            # Matrix is used when chi does not exist.
+            # We have to check that the correct form is given, but only if
+            # chi already has been set.  If not, we just check that this is OK
+            # and set the use_chi flag accordingly
+
+            # First lets set our dimensions here since they get used repeatedly
+            # throughout this code.
+            if self._representation is 'isotropic':
```

Comment:
This can leverage a new `shape` property as well, with a simple concatenation to the tuple returned from the property in the case of a fission matrix.